core/server/services/auth/api-key/content.js

```diff
@@ -0,0 +1,37 @@
+const models = require('../../../models');
+const common = require('../../../lib/common');
+
+const authenticateContentApiKey = function authenticateContentApiKey(req, res, next) {
+    // allow fallthrough to other auth methods or final ensureAuthenticated check
+    if (!req.query || !req.query.content_key) {
+        return next();
+    }
+
+    let key = req.query.content_key;
+
+    models.ApiKey.findOne({secret: key}).then((apiKey) => {
```

Comment:
This feels like a controller.

Reply:
Is that because it uses a model? This deffo isn't a controller, IMO, we have other middlware that does similar stuff. e.g. session middleware use model to get session out of db, we have a middleware to populate req.user with model, and the oauth2 middleware would populate oauth clients from the model. 

Let me know if you still think this feels like a controller, I would be confused as to how it would working using a controller as middleware though :thinking:

Reply:
Yeah it isn't. I thought it's a single route, but it's used for any route. lalaa

Reply:
:notes: lalaa